### ISSUE TITLE
Use releases and release notes

### ISSUE THREAD

**Issue by ysmoradi** [simple-odata-client/Simple.OData.Client#846]
I've generated the following image using [Github auto-generate release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) feature.
I think it would be nice to provide release notes to our community

![image](https://user-images.githubusercontent.com/5352980/145723506-68a739d5-0b41-4168-ac1b-ea9795c83623.png)

**Comment by davidnmbond** [comment]
Yes - looks nice, and we we should also ensure that the release notes match what gets presented in the nuget release notes field in the nuspec / csproj.
